clconf/template.go

```diff
@@ -0,0 +1,221 @@
+package clconf
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+const setuid = 4
+const setgid = 2
+const sticky = 1
+
+// TemplateOptions are settings for ProcessTemplates.
+type TemplateOptions struct {
+	// CopyTemplatePerms uses the existing template permissions instead of FileMode for template
+	// permissions
+	CopyTemplatePerms bool
+	// Flatten flattens the templates into the root of the dest instead of the preserving
+	// the relative path under the source.
+	Flatten bool
+	// KeepExistingPerms determines whether to use existing permissions on template files that are overwritten.
+	KeepExistingPerms bool
+	// Rm determines whether template files distinct from their target are deleted after processing.
+	Rm bool
+	// FileMode is the permissions to apply to template files when writing.
+	FileMode os.FileMode
+	// DirMode is the permission similar to FileMode but for new folders.
+	DirMode os.FileMode
+	// Extension is the extension to use when searching folders. If missing all files will be used.
```

Comment:
More pedantic commentary here, but _missing_ seems like the wrong word...  It cant really _be missing_ as there is no such thing.  Given that there is no such thing, the only reasonable interpretation of missing is has value of empty string, so i guess it doesn't matter.